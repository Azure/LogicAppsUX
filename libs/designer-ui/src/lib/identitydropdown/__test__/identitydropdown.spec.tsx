import { setIconOptions } from '@fluentui/react';
import renderer from 'react-test-renderer';
import { IdentityDropdown, IdentityDropdownProps } from '../index';

describe('lib/identitydropdown', () => {
  let minimal: IdentityDropdownProps;

  beforeAll(() => {
    setIconOptions({
      disableWarnings: true,
    });
  });

  beforeEach(() => {
    minimal = {
      defaultSelectedKey: 'key1',
      dropdownOptions: [{ key: 'key1', text: 'text1' }],
      handleChange: jest.fn(),
      readOnly: false,
    };
  });

  it('should render', () => {
    const tree = renderer.create(<IdentityDropdown {...minimal} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
