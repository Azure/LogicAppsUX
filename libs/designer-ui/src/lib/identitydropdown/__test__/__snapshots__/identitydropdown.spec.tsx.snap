// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/identitydropdown should render 1`] = `
<div
  className="msla-identity-dropdown-container"
>
  <label
    className="ms-Label msla-identity-dropdown-label root-109"
  >
    Managed identity
  </label>
  <div
    className="ms-Dropdown-container"
  >
    <div
      aria-disabled={false}
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Managed identity"
      className="ms-Dropdown dropdown-110"
      data-is-focusable={true}
      data-ktp-target={true}
      id="Dropdown0"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      <span
        aria-invalid={false}
        className="ms-Dropdown-title ms-Dropdown-titleIsPlaceHolder title-111"
        id="Dropdown0-option"
      >
        Select a managed identity
      </span>
      <span
        className="ms-Dropdown-caretDownWrapper caretDownWrapper-112"
      >
        <i
          aria-hidden={true}
          className="ms-Dropdown-caretDown caretDown-126"
          data-icon-name="ChevronDown"
        />
      </span>
    </div>
  </div>
</div>
`;
